logfire/_internal/profiling/supervisor.py

```
@@ -0,0 +1,232 @@
+
+        The profiler exits 0 even when it cannot read the target's memory, so
+        success is judged by whether non-empty output was actually written.
+        """
+        with tempfile.TemporaryDirectory(prefix='logfire-profiling-') as tmp:
+            out = Path(tmp) / 'chunk.collapsed'
+            try:
+                with self._lock:
+                    if self._stop.is_set():
+                        return None  # shutting down: don't start a profiler nobody will terminate
+                    cmd = _spawn_command(self._profiler_command(out, pid, duration, rate))
+                    self._proc = proc = subprocess.Popen(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)